src/components/DatePicker/DatePicker.tsx

```diff
@@ -47,22 +48,26 @@ export default function DatePicker(props: Props): JSX.Element {
     </>
   );
 
+  // set default timezone to UTC
+  moment.tz.setDefault('UTC');
+
   return (
     <div className={`date-picker ${props.className} ${props.errorText ? 'date-picker--error' : ''}`}>
-      <LocalizationProvider dateAdapter={AdapterDateFns}>
+      <LocalizationProvider dateAdapter={AdapterMoment} dateLibInstance={moment}>
         {props.label && (
           <label htmlFor={props.id} className='textfield-label'>
             {props.label}
           </label>
         )}
         <DesktopDatePicker
           onError={props.onError}
-          minDate={props.minDate}
-          maxDate={props.maxDate}
-          inputFormat='yyyy-MM-dd'
-          value={props.value ? moment(props.value, 'YYYY-MM-DD').toDate() : null}
-          onChange={(newValue: string | null) => {
-            props.onChange(props.id, newValue);
+          minDate={props.minDate ? moment(props.minDate) : undefined}
+          maxDate={props.maxDate ? moment(props.maxDate) : undefined}
+          inputFormat='yyyy-MM-DD'
+          value={temporalValue}
+          onChange={(newValue: any) => {
+            setTemporalValue(newValue);
+            props.onChange(props.id, newValue?.isValid() ? newValue?.toDate() : null);
```

Comment:
So we don't change the value if it's not valid, right? Why did you say that client will decide that?

Reply:
we send back null to the client if it's not valid - this is the closest we can get to backwards compatibility because moment passes a date object even if date is not correct. 
But moment.isValid() will catch that.